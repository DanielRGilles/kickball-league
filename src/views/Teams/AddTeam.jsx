import { useState } from 'react'
import { useHistory } from 'react-router-dom'

export default function AddTeam() {
    const [ name, setName ] = useState()
    const [ city, setCity ] = useState()
    const [ state, setState ] = useState()
    const history = useHistory()

    const handleSubmit = async (e) => {
        e.preventDefault();
        const resp = await createTeam( { name, city, state});
         history.push(`/teams/${resp[0].id}`)
    }
    return (
        <>
         <fieldset>
        <legend>Add Team</legend>     
        <form onSubmit={handleSubmit}>
            <label>Name</label>
            <input id='name' name='name' type="text" />
            <label>City</label>
            <input type="text" />
            <label>State</label>
            <input type="text" />
            <button type='submit'>Add Team </button>
        </form>
        </fieldset>   
        </>
    )
}
