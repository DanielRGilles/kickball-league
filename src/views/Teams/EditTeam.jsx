import { useEffect, useState } from 'react'
import { useHistory, useParams } from 'react-router-dom'
import { updateTeamById, getTeamById } from'../../services/teams.jsx'

export default function EditTeam() {
    const [ name, setName ] = useState()
    const [ city, setCity ] = useState()
    const [ state, setState ] = useState()
    const history = useHistory()
    const { id } = useParams();
    const [ team, setTeam ] = useState();
    const [ loading, setLoading ] = useState(true);
    
    const loadTeam = async () => {
        setLoading(true);
        const resp = await getTeamById(id);
        setTeam(resp);
        setLoading(false);
      };
    useEffect(() => {
         loadTeam();
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [] )
    

    async function handleSubmit(e) {
        e.preventDefault()
        const resp = await updateTeamById(id, { name, city, state })
        history.push(`/teams/${resp[0].id}`)
    }

    if (loading) return <h1>Loading teams...</h1>
    return (
        <>
         <fieldset>
        <legend>Add Team</legend>     
        <form onSubmit={handleSubmit}>
            <label htmlFor='name' >Name:</label>
            <input id='name' placeholder={team.name} name='name' type="text" value={name} onChange={(e) => setName(e.target.value)} />
            <label htmlFor='city' >city:</label>
            <input id='city' placeholder={team.city} name='city' type="text" value={city} onChange={(e) => setCity(e.target.value)} />
            <label htmlFor='state' >state:</label>
            <input id='state' placeholder={team.state} name='state' type="text" value={state} onChange={(e) => setState(e.target.value)} />
           
            <button type='submit'>Edit Team </button>
        </form>
        </fieldset>   
        </>
    )
}
