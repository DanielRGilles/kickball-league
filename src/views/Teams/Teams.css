li {
    
    list-style: none;
}
.player {
    padding-left: 15px;
    font-weight: 700;
}