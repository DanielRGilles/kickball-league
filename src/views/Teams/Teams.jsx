import React from 'react'
import { useState, useEffect } from 'react'
import { getTeams, deleteTeamById } from '../../services/teams.jsx'
import { Link } from 'react-router-dom'

export default function Teams() {
    const [ teams, setTeams ] = useState([])
    const [ loading, setLoading ] = useState(true)

    const loadTeams = async () => {
        setLoading(true);
        const resp = await getTeams();
        setTeams(resp);
        setLoading(false);
      };

    useEffect(() => {
        loadTeams();
    }, [] )

    const handleDelete = async ({ id, name }) => {
            await  deleteTeamById(id);
            alert(`You have deleted ${name} from the team list`)
            await loadTeams();
        }
           
            
    
    
    if (loading) return <h1>Loading teams...</h1>
    return (
        <>
        <button><Link to='/teams/new' >Add a Team</Link></button>
        <ul>
            {teams.map((team) => (
                <li key={team.id}><Link to={`/teams/${team.id}`}>{team.name}</Link><button ><Link to={`/teams/${team.id}/edit`}> Edit Team</Link> </button><button id={team.id} onClick={() => handleDelete({ id: team.id, name: team.name })}>Delete Team</button></li>
            ))}
            
        </ul>
        </>
    )
}
