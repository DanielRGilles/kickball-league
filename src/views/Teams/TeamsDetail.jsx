import { useEffect, useState } from 'react'
import { useParams } from 'react-router-dom'
import { getTeamById } from '../../services/teams.jsx'


export default function TeamsDetail() {
    const { id } = useParams()
    const [ team , setTeam ] = useState()
    const [ loading , setLoading ] = useState(true)
    
    useEffect(() => {
        getTeamById(id)
        .then((data) => setTeam(data))
        .catch((error) => console.error(error))
        .finally(() => setLoading(false))
    }, [id] )

    if (loading) return <h1>Loading team...</h1>
    return (
            <ul> {console.log(team)}
               <li>{team.name}</li>
               <li>{team.city}</li>
               <li>{team.state}</li>
               {team.players.map((player) => <li key={player.id}>{player.name}</li>)}
           
            </ul>
            
    )
}
