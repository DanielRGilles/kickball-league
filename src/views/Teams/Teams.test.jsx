import { render } from '@testing-library/react'

import Teams from './Teams'

it('Should fetch then render a list of teams', async () => {
  const { container } = render(<Teams />)
  expect(container).toMatchSnapshot()
})
