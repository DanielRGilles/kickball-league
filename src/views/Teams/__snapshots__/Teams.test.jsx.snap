// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should fetch then render a list of teams 1`] = `
<div>
  <h1>
    Loading teams...
  </h1>
</div>
`;
